src/interface/pipe.rs

```diff
@@ -14,17 +20,42 @@ use std::slice;
 use std::sync::atomic::{AtomicBool, AtomicU32, Ordering};
 use std::sync::Arc;
 
+// lets define a few constants for permission flags and the standard size of a Linux page
 const O_RDONLY: i32 = 0o0;
 const O_WRONLY: i32 = 0o1;
 const O_RDWRFLAGS: i32 = 0o3;
 const PAGE_SIZE: usize = 4096;
 
-const CANCEL_CHECK_INTERVAL: usize = 1048576; // check to cancel pipe reads every 2^20 iterations
-
+// lets also define an interval to check for thread cancellations since we may be blocking in trusted space here
+// we're going to define this as 2^20 which should be ~1 sec according to the standard CLOCKS_PER_SEC definition, though it should be quite shorter on modern CPUs
+const CANCEL_CHECK_INTERVAL: usize = 1048576;
```

Comment:
I'm confused about this.  What is this actually for?  Are you busy waiting?  

(Also please line wrap the comments to 80 chars)

Reply:
Yes we have to check for cancellation when we busy wait, this is in read specifically